### ISSUE TITLE
Update operations limited to max 50k

### ISSUE THREAD

**Issue by szibis** [go-graphite/go-carbon#32]
With normal traffic i get max 50k on update operation whatever i set in go-carbon config.

There are two situations when it is much higher:
- when i set max updates to unlimited
- when i stop go-carbon and it's start flushing to drives

I think there is problem with throttling in go-carbon. Any ideas ?

On unlimited setting drives handling traffic with 200k+ update with no problem, but i use too much system cpu, drives lifetime is going down on SSD fast and i would like to have option of tuning IOPS<->Cache usage with more precision.

```
[common]
user = "carbon"
logfile = "/var/log/go-carbon/go-carbon.log"
log-level = "debug"
graph-prefix = "carbon.agents.{host}."
max-cpu = 15
metric-interval = "10s"

[whisper]
data-dir = "/opt/graphite/storage/whisper/"
schemas-file = "/etc/go-carbon/storage-schemas.conf"
aggregation-file = "/etc/go-carbon/storage-aggregation.conf"
workers = 32
max-updates-per-second = 75000
enabled = true

[cache]
max-size = 10000000
input-buffer = 500000

[udp]
listen = ":2003"
enabled = true
log-incomplete = true

[tcp]
listen = ":2003"
enabled = true

[pickle]
listen = ":2005"
enabled = false

[carbonlink]
listen = "127.0.0.1:7002"
enabled = true
read-timeout = "30s"
query-timeout = "1s"

[pprof]
listen = "localhost:7007"
enabled = false
```

```
[2016-01-15 12:34:14] I [persister] doCheckpoint() {commitedPoints=324155, created=0, updateOperations=39988}
[2016-01-15 12:34:24] I [persister] doCheckpoint() {commitedPoints=694713, created=0, updateOperations=18599}
[2016-01-15 12:34:34] I [persister] doCheckpoint() {commitedPoints=1569877, created=0, updateOperations=33807}
[2016-01-15 12:34:44] I [persister] doCheckpoint() {commitedPoints=1618408, created=0, updateOperations=34857}
[2016-01-15 12:34:54] I [persister] doCheckpoint() {commitedPoints=1157205, created=0, updateOperations=40795}
[2016-01-15 12:35:04] I [persister] doCheckpoint() {commitedPoints=1076430, created=0, updateOperations=44264}
[2016-01-15 12:35:14] I [persister] doCheckpoint() {commitedPoints=994659, created=0, updateOperations=46762}
[2016-01-15 12:35:24] I [persister] doCheckpoint() {commitedPoints=965929, created=0, updateOperations=46725}
[2016-01-15 12:35:34] I [persister] doCheckpoint() {commitedPoints=550507, created=0, updateOperations=24489}
[2016-01-15 12:35:44] I [persister] doCheckpoint() {commitedPoints=1013312, created=0, updateOperations=38711}
[2016-01-15 12:35:54] I [persister] doCheckpoint() {commitedPoints=1003661, created=0, updateOperations=38305}
[2016-01-15 12:36:04] I [persister] doCheckpoint() {commitedPoints=786854, created=0, updateOperations=39359}
[2016-01-15 12:36:14] I [persister] doCheckpoint() {commitedPoints=730721, created=0, updateOperations=41951}
[2016-01-15 12:37:26] I [persister] doCheckpoint() {commitedPoints=305571, created=0, updateOperations=31096}
[2016-01-15 12:37:36] I [persister] doCheckpoint() {commitedPoints=1228619, created=0, updateOperations=38687}
[2016-01-15 12:37:46] I [persister] doCheckpoint() {commitedPoints=1439732, created=0, updateOperations=26298}
[2016-01-15 12:37:56] I [persister] doCheckpoint() {commitedPoints=1493276, created=0, updateOperations=24295}
[2016-01-15 12:38:06] I [persister] doCheckpoint() {commitedPoints=1835976, created=0, updateOperations=33459}
[2016-01-15 12:38:16] I [persister] doCheckpoint() {commitedPoints=1144794, created=0, updateOperations=25530}
[2016-01-15 12:38:26] I [persister] doCheckpoint() {commitedPoints=1005914, created=0, updateOperations=25490}
[2016-01-15 12:38:36] I [persister] doCheckpoint() {commitedPoints=1279118, created=0, updateOperations=40149}
[2016-01-15 12:38:46] I [persister] doCheckpoint() {commitedPoints=1217856, created=0, updateOperations=43423}
[2016-01-15 12:38:56] I [persister] doCheckpoint() {commitedPoints=1009972, created=0, updateOperations=45512}
[2016-01-15 12:39:06] I [persister] doCheckpoint() {commitedPoints=947630, created=0, updateOperations=41562}
[2016-01-15 12:39:16] I [persister] doCheckpoint() {commitedPoints=771803, created=0, updateOperations=32007}
[2016-01-15 12:39:26] I [persister] doCheckpoint() {commitedPoints=887938, created=0, updateOperations=31423}
```

**Comment by szibis** [comment]
I manage this by config changes:

```
workers = 6
max-updates-per-second = 0
enabled = true

[cache]       [cache]
max-size = 20000000
input-buffer = 512000
```

With fast SSD drives on backend i would like to get power of storage, but with control of system/user(go-carbon) CPU usage and cache size/flush intervals. 

Now it is much better in my case.

**Comment by lomik** [comment]
The problem really exists. Throttling performance will be fixed in next version

**Comment by szibis** [comment]
Ok, that is good news. Thanks.

**Comment by lomik** [comment]
Released in 0.7
